app/src/main/java/com/github/geohunt/app/ui/components/profile/edit/EditProfileContent.kt

```diff
@@ -1,51 +1,31 @@
 package com.github.geohunt.app.ui.components.profile.edit
 
-import android.widget.Toast
+import android.net.Uri
 import androidx.compose.foundation.layout.Column
 import androidx.compose.material.icons.Icons
 import androidx.compose.material.icons.filled.Pending
 import androidx.compose.material.icons.filled.Save
 import androidx.compose.runtime.*
 import androidx.compose.ui.Modifier
-import androidx.compose.ui.platform.LocalContext
 import androidx.compose.ui.platform.testTag
 import androidx.compose.ui.res.stringResource
-import com.github.geohunt.app.model.database.Database
-import com.github.geohunt.app.model.database.api.User
 import com.github.geohunt.app.ui.components.button.FlatLongButton
-import com.github.geohunt.app.utility.findActivity
 import com.github.geohunt.app.R
-import com.github.geohunt.app.model.database.api.EditedUser
+import com.github.geohunt.app.model.EditedUser
+import com.github.geohunt.app.model.User
 
 @Composable
-fun EditProfileContent(user: User) {
-    instrumentableEditProfileContent(user = user)
-}
-
-@Composable
-@NoLiveLiterals
-fun instrumentableEditProfileContent(user: User): MutableState<EditedUser> {
-    val editedUser = remember { mutableStateOf(EditedUser.fromUser(user)) }
-    val db = Database.databaseFactory.get()(LocalContext.current.findActivity())
-    var isSaving by remember { mutableStateOf(false) }
-    val ctx = LocalContext.current
-
-    fun save() {
-        isSaving = true
-        val newUser = editedUser.value
-        db.updateUser(newUser)
-            .addOnCompleteListener { isSaving = false }
-            .addOnFailureListener {
-                Toast.makeText(ctx, "Error while updating profile.", Toast.LENGTH_LONG).show()
-                isSaving = false
-            }
-
-    }
-
+fun EditProfileContent(
+    user: User,
+    eu: EditedUser,
```

Comment:
Could we just past the view model down here instead on relying on the EditedUser ?